openapi/dcos.yaml

```diff
@@ -2376,8 +2376,339 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/MetronomeV1Error"
+  /service/metronome/v1/jobs/{jobId}/schedules:
+    get:
+      description: Get the list of all schedules for this jobId.
+      operationId: V1GetJobSchedules
```

Comment:
The V1 prefix is interesting here, at the moment in the spec we only version the model schemas.

I think for consistency we should omit it and have methods like `GetJobSchedules`. We had some discussions in the past on if we want to support multiple versions of APIs, and thought that it is mostly relevant for data services, and for DC/OS components we would stick to a single version. We would then create a new branch and update models accordingly each time a new DC/OS minor version is released.

It's definitely an interesting point though and if we want to improve multi-versions it in the future and add version prefixes to operation methods it should be tackled globally to make sure all the components spec are consistent.

Reply:
What if `v2` of the API comes out and there are object changes; we'd then have to do this anyways?

Reply:
@bamarni we do the same with edgelb 

https://github.com/dcos/client-go/blob/master/openapi/dcos.yaml#L2104-L2107

Metronome has currently two active version v0 and v1. To me it makes sense to distinguish. We could argue about the convention of `<version><op-name>` but in that case we also need to change edgelb

Reply:
From what I've understood from edgelb, the 2 versions are meant to be served simultaneously and are defined in the same spec (see https://docs.mesosphere.com/services/edge-lb/1.3/api-reference/edgelb/#api-versions) : 

> The V1 and V2 specifications were merged into a single spec; however, there are still separate v1 and v2 docs for reference configs, pool examples, etc.

Thus I reflected in the client-go specs too. I think the Metronome use-case is more similar to Secrets or IAM, which don't use the V1 prefix.

Again, I'm not against this change, I just think consistency is the most important. And if we change it we should do it for all components and it deserves a separate PR.

Reply:
Are we OK leaving it how it is then, or?

Reply:
Looking at https://dcos.github.io/metronome/docs/generated/api.html, it seems actually similar to edgelb, with not versioned endpoints like `/ping` and then `/v0` or `/v1` which are served together. So that's fine to keep it as is :+1:

Reply:
perfect... i'll resolve this conversation